srv_hijacker/srv_hijacker.py

```diff
@@ -78,3 +127,21 @@ def hijack(host_regex, srv_dns_host=None, srv_dns_port=None):
         srv_resolver.port = int(srv_dns_port)
 
     socket.getaddrinfo = patched_socket_getaddrinfo(host_regex, srv_resolver)
+
+    if patch_everything:
+        libs = {each: True for each in SUPPORTED_LIBS.keys()}
+
+    for k, v in libs.items():
+        if k not in SUPPORTED_LIBS:
+            raise ValueError(f"library {k} is not supported")
+
+        if not v:
+            continue
+
+        try:
+            SUPPORTED_LIBS.get(k)(host_regex, srv_resolver)
+        except BaseException as e:
```

Comment:
I'm not sure how `BaseException` is different from `Exception` - I'm used to seeing just `Exception` in code of this sort, is there a specific reason to use `BaseException` here?